add task solution

StanislavKapytsia opened the pull request:
- [DEMO LINK](https://StanislavKapytsia.github.io/js_promise_basic_DOM/)

Create 2 promises `promise1` and `promise2`.
- `promise1` should be resolved on a `click` on `.logo`.
- `promise2` promise should be rejected in 3 seconds.
- add success and error handlers for both promises
- success handler should append a `<div class="message">` with the text `Promise was resolved!` to the body
- error handler should append a `<div class="message error-message">` with the text `Promise was rejected!` to the body

StanislavKapytsia committed:
add task solution

etojeDenys commented on src/scripts/main.js:
```diff
@@ -1,3 +1,56 @@
+};
+
+const promise1 = new Promise((resolve, reject) => {
+  logo.onclick = (e) => {
```
in js, we usually use the addEventListener instead of the onClick

volodymyr-soltys97 replied:
Still not fixed 
<img width="908" alt="image" src="https://github.com/user-attachments/assets/83a4859b-fde2-4b9a-9605-0205762064b1">

etojeDenys commented on src/scripts/main.js:
```diff
@@ -1,3 +1,56 @@
+
+message.classList.add('message');
+
+const badResult = function (content) {
+  message.classList.add('error-message');
+
+  message.textContent = content;
+  body.append(message);
+
+  return message;
+};
+
+const goodResult = function (content) {
+  message.classList.toggle('error-message', false);
+  message.textContent = content;
+  body.append(message);
+
+  return message;
+};
```
you need to create the message in this function instead of changing already created element, also try to combine these functions into one

StanislavKapytsia committed:
add task solution 1.0

StanislavKapytsia committed:
Update main.js

volodymyr-soltys97 reviewed (changes requested):
Hi, you need to fix all comments from the previous review

StanislavKapytsia committed:
fix main.js

add addEventListener
fix small details

etojeDenys commented on src/scripts/main.js:
```diff
@@ -1,3 +1,61 @@
+  logo.addEventListener('click', (e) => {
+    logo.dataset.click = 'click';
+
+    if (logo.hasAttribute('data-click')) {
+      if ('addInfo' in data) {
+        delete data.addInfo;
+      }
+
+      data.message = `Promise was resolved!`;
+      resolve(data);
+    } else {
+      data.message = `Promise was rejected!`;
+      data.addInfo = 'error-message';
+
+      reject(data);
+    }
```
why do you need this logic, you can just create the object in the resolve
```
resolve({
 ...........................
})
```
or in reject

StanislavKapytsia replied:
![Знімок екрана 2024-09-17 165700](https://github.com/user-attachments/assets/4ca44b8f-8bff-40b9-9d63-e44353a83bdb)

StanislavKapytsia replied:
![Знімок екрана 2024-09-17 165738](https://github.com/user-attachments/assets/f551ebb6-1d1a-4d9d-9b92-f482798cc002)

StanislavKapytsia replied:
there is something problem with All checks have failed two 
the same code one ok another not

StanislavKapytsia replied:
![Знімок екрана 2024-09-17 173508](https://github.com/user-attachments/assets/73f584a7-8f4c-49a3-aba1-12a1627b388a)
![Знімок екрана 2024-09-17 173550](https://github.com/user-attachments/assets/9b7e948f-e0f1-449b-94bf-ac3b3f0c45b5)

StanislavKapytsia committed:
change condition.js

 if (logo.hasAttribute('data-click')) {
      resolve({
        message: `Promise was rejected!`,
      });
    } else {
      const error = {
        message: `Promise was rejected!`,
        addInfo: 'error-message',
      };

      reject(error);
    }

StanislavKapytsia committed:
Update main.js

check condition
